### PR TITLE
fix: stricter empty block check

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

#647 introduced a new bug in which some blocks are being stripped out of the page while being parsed by its filters. This is due the filtering out of blocks without `innerHTML`.

Not all blocks require `innerHTML` to display content. Blocks such as the Donation block and the Homepage Posts block uses other attributes to render their content.

This PR implements a stricter check to only filter out empty innerHTML of the following blocks:

- `null`
- core/paragraph
- core/heading
- core/list
- core/quote
- core/html
- core/freeform

It also implements a new assertion for more readable and less sensitive testing of block content insertion.

### How to test the changes in this Pull Request:

1. On the master branch, make sure you have inline campaigns visible
2. Create a page with donation and homepage posts block
3. Visit the page and see the blocks are not visible
4. Check out this branch and confirm they are displayed as expected

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by adekbadek — COMMENTED** [review]
How about getting rid of that check instead of making it smarter? There might be other blocks (by other plugins) this can seem empty, we’ll never know. 
If a block is empty, it's 0-length content won't throw off the insertion computation anyway.

**Comment by miguelpeixe** [comment]
a5e11fa71f303167273b3f44316dd6c38b70dd04 restores the check including `null` and 8827fbfd6939a7c54960fe453c5f6db2d1ea630d implements a new assertion for more readable and less sensitive testing of block content insertion.

File: includes/class-newspack-popups-inserter.php [hunk 1]
```diff
@@ -219,12 +219,24 @@ public static function insert_popups_in_post_content( $content, $popups ) {
 
 		$parsed_blocks = self::convert_classic_blocks( parse_blocks( $content ) );
 
-		$parsed_blocks = array_values( // array_values will reindex the array.
+		// List of blocks that require innerHTML to render content.
+		$innerhtml_blocks_names = [
```
Comment (adekbadek) [inline includes/class-newspack-popups-inserter.php:223]: The comment and variable name are a bit unclear to me. This is effectively a list of blocks which render nothing if their `innerHTML` is empty. While I don't have a great idea for the variable name, I think the comment does not reflect the intention here.

File: tests/test-content-intertion.php [hunk 2]
```diff
@@ -35,6 +35,18 @@ public static function rendered_popup( $id ) {
 		return '<!-- wp:shortcode -->[newspack-popup id="' . $id . '"]<!-- /wp:shortcode -->';
 	}
 
+	/**
+	 * Assert that that serialized blocks match the block names.
```
Comment (adekbadek) [inline tests/test-content-intertion.php:39]:
```suggestion
	 * Assert that serialized blocks match the block names.
```

**Review by adekbadek — APPROVED** [review]
Just some naming/comment nitpicks 👌

File: includes/class-newspack-popups-inserter.php (hunk 1 above)
Comment (miguelpeixe) [reply includes/class-newspack-popups-inserter.php]: I agree! What do you think of 93785c75ff2a9f3f9ea6be68dabb823df51d1016 ?

**Comment by miguelpeixe** [comment]
For documentation purposes, we've restored the empty block check due to its importance on the _do-not-insert-after-block_ logic. Empty blocks passing through would visually disrupt the purpose of the logic, making it seem like a block is inserted after the restricted block when it's not.

**Review by adekbadek — APPROVED** [review]

**Pull request merged**